Library/ViewModels/PaymentMethodsViewModel.swift

```diff
@@ -24,17 +28,52 @@ public final class PaymentMethodsViewModel: PaymentMethodsViewModelType,
 PaymentMethodsViewModelInputs, PaymentMethodsViewModelOutputs {
 
   public init() {
-
     let paymentMethodsEvent = self.viewDidLoadProperty.signal
       .switchMap { _ in
         AppEnvironment.current.apiService.fetchGraphCreditCards(query: UserQueries.storedCards.query)
           .ksr_delay(AppEnvironment.current.apiDelayInterval, on: AppEnvironment.current.scheduler)
           .materialize()
-      }
+    }
+
+    let deletePaymentMethodEvents = self.didDeleteCreditCardSignal.switchMap { creditCard in
+      AppEnvironment.current.apiService.deletePaymentMethod(input: .init(paymentSourceId: creditCard.id))
+        .ksr_delay(AppEnvironment.current.apiDelayInterval, on: AppEnvironment.current.scheduler)
+        .materialize()
+    }
+
+    let deletePaymentMethodEventsErrors = deletePaymentMethodEvents.errors()
+
+    self.showAlert = deletePaymentMethodEventsErrors
+      .ignoreValues()
+      .map {
+        localizedString(
+          key: "Something_went_wrong_and_we_were_unable_to_remove_your_credit_card_please_try_again",
+          //swiftlint:disable:next line_length
+          defaultValue: "Something went wrong and we were unable to remove your credit card, please try again."
+        )
+    }
 
-    self.paymentMethods = paymentMethodsEvent.values().map { $0.me.storedCards.nodes }
+    let paymentMethodsValues = paymentMethodsEvent.values().map { $0.me.storedCards.nodes }
+
+    self.paymentMethods = Signal.merge(
+      paymentMethodsValues,
+      paymentMethodsValues.takeWhen(deletePaymentMethodEventsErrors)
+    )
 
     self.goToAddCardScreen = self.didTapAddCardButtonProperty.signal
+
+    self.tableViewIsEditing = self.editButtonTappedSignal.scan(false) { current, _ in !current }
+  }
+
+  let (didDeleteCreditCardSignal, didDeleteCreditCardObserver) = Signal<GraphUserCreditCard.CreditCard,
+    NoError>.pipe()
+  public func didDelete(_ creditCard: GraphUserCreditCard.CreditCard) {
+    self.didDeleteCreditCardObserver.send(value: creditCard)
+  }
+
+  let (editButtonTappedSignal, editButtonTappedObserver) = Signal<(), NoError>.pipe()
```

Comment:
Would really like to start using `pipe`s instead of `MutableProperty`s. `pipe()` has the following signature:

```swift
func pipe(disposable: Disposable? = nil) -> (output: Signal, input: Observer)
```

This allows us to `send` values to the `input` which is an `Observer`. The main benefit over a `MutableProperty` is that it does not require an initial value, and therefore doesn't require an initial `nil` value as we tend to provide. That then in turn means that we don't need to use `.signal.skipNil()` everywhere because it's already a `Signal<T, E>`.

Reply:
This approach is actually more used for what I've seen. I don't see a problem with it, just think the team should talk about it to avoid using to different approaches to do the same thing. Maybe it's a good opportunity for a rfc? 😄

Reply:
I don't think it's a big enough change for an RFC, will just discuss at next guild 👍